src/main/java/com/khalsa_ji/ems/rest/Post.java

```diff
@@ -0,0 +1,137 @@
+                    .body(messageSource.getMessage("error.code.nullString", JOB_TITLE, Locale.getDefault()));
+            case emptyString:       return ResponseEntity.badRequest()
+                    .body(messageSource.getMessage("error.code.emptyString", JOB_TITLE, Locale.getDefault()));
+            case invalidString:     return ResponseEntity.badRequest()
+                    .body(messageSource.getMessage("error.code.invalidString", JOB_TITLE, Locale.getDefault()));
+        }
+
+        //  Validating Manager ID
+        switch(Validator.validateID(employee.getManagerID())) {
+            case nullNumber:        return ResponseEntity.badRequest()
+                    .body(messageSource.getMessage("error.code.nullNumber", MANAGER, Locale.getDefault()));
+            case zero:              return ResponseEntity.badRequest()
+                    .body(messageSource.getMessage("error.code.zero", MANAGER, Locale.getDefault()));
+        }
+
+        Designation designation = designationService.getDesignation(employee.getJobTitle());
+
+        //  If designation with specified id does not exists.
+        if(designation.getLevelID() == -1)
+            return ResponseEntity.badRequest()
+                    .body(messageSource.getMessage("error.designation.notFound",
+                            new String[]{employee.getJobTitle()}, Locale.getDefault()));
+
+        employee.setJobID(designation);
+
+        Employee manager = employeeService.getEmployeeByID(employee.getManagerID());
+
+        //  If employee's job title(designation) is not found to be Director
+        if(!employee.getJobTitle().equals("Director")) {
+            //  If manager id found to be invalid
+            if(manager.getEmployeeID() == 0)
+                return ResponseEntity.badRequest()
+                        .body(messageSource.getMessage("error.employee.manager.notFound",
+                                new Long[]{employee.getManagerID()}, Locale.getDefault()));
+
+            //  If employee possesses a higher designation than its manager
+            if(manager.getJobID().getLevelID() >= employee.getJobID().getLevelID())
+                return ResponseEntity.badRequest()
+                        .body(messageSource.getMessage("error.employee.jobID.higherThan.manager",
+                                null, Locale.getDefault()));
+        }
+
+        //  If employee's job title(designation) is found to be Director and there already exists another Director
+        if(designation.getDesignation().equals("Director") && employeeService.getDirector().getEmployeeID() != 0)
+            return ResponseEntity.badRequest()
+                    .body(messageSource.getMessage("error.director.alreadyExists", null, Locale.getDefault()));
+
+        employeeService.addEmployee(employee);
+
+        URI location = ServletUriComponentsBuilder
+                .fromCurrentRequest().path("/{id}")
+                .buildAndExpand(employee.getEmployeeID()).toUri();
```

Comment:
I think, this URI was created to support UI. 
Am i correct ?

Reply:
URI is helpful for both Frontend as well as Backend.

According to the **IETF - RFC 7231**, it is a best practice(standard) to return HttpStatus code of 201(created) on successfully saving the new requested resource using _POST_.
In most of the cases, the server is responsible for generating a unique ID for that resource. So, why to send the whole object, which just got created to the user.
As it would be a overhead for the _POST_ request also in some or the other way. Instead, just send the new _URI_ generated, which is unique by its nature, to the user in the form of a field - _Location_ inside the HTTP header.
If in case any user or frontend wants to view/display the contents can do so by making a _GET_ request to that _URI_.

References:
- POST:
   [https://tools.ietf.org/html/rfc7231#section-4.3.3](https://tools.ietf.org/html/rfc7231#section-4.3.3)
- Location: HTTP header field,
   [https://tools.ietf.org/html/rfc7231#section-7.1.2](https://tools.ietf.org/html/rfc7231#section-7.1.2)